pkg/sources/git/git.go

```diff
@@ -329,7 +361,15 @@ func CloneRepo(ctx context.Context, userInfo *url.Userinfo, gitURL string, args
 	if err = GitCmdCheck(); err != nil {
 		return "", nil, err
 	}
-	clonePath, err := os.MkdirTemp(os.TempDir(), "trufflehog")
+
+	pid, err := GetScannerPIDs(ctx, ExecutableName)
+	if err != nil {
+		return "", nil, err
+	}
+
+	tmpdir := fmt.Sprintf("%s-%d-", ExecutableName, pid)
```

Comment:
I recommend adding a MkdirTemp to a `cleantemp` package that takes care of all of this, using os.Getpid instead of GetScannerPIDs